homeassistant/components/hlk_sw16/__init__.py

```diff
@@ -2,139 +2,125 @@
 import logging
 
 from hlk_sw16 import create_hlk_sw16_connection
-import voluptuous as vol
-
-from homeassistant.const import (
-    CONF_HOST,
-    CONF_NAME,
-    CONF_PORT,
-    CONF_SWITCHES,
-    EVENT_HOMEASSISTANT_STOP,
-)
+
+from homeassistant.const import CONF_HOST, CONF_PORT
 from homeassistant.core import callback
-import homeassistant.helpers.config_validation as cv
-from homeassistant.helpers.discovery import async_load_platform
 from homeassistant.helpers.dispatcher import (
     async_dispatcher_connect,
     async_dispatcher_send,
 )
 from homeassistant.helpers.entity import Entity
 
+from .const import (
+    CONNECTION_TIMEOUT,
+    DEFAULT_KEEP_ALIVE_INTERVAL,
+    DEFAULT_RECONNECT_INTERVAL,
+    DOMAIN,
+)
+
 _LOGGER = logging.getLogger(__name__)
 
 DATA_DEVICE_REGISTER = "hlk_sw16_device_register"
-DEFAULT_RECONNECT_INTERVAL = 10
-DEFAULT_KEEP_ALIVE_INTERVAL = 3
-CONNECTION_TIMEOUT = 10
-DEFAULT_PORT = 8080
+DATA_DEVICE_LISTENER = "hlk_sw16_device_listener"
 
-DOMAIN = "hlk_sw16"
-
-SWITCH_SCHEMA = vol.Schema({vol.Optional(CONF_NAME): cv.string})
-
-RELAY_ID = vol.All(
-    vol.Any(0, 1, 2, 3, 4, 5, 6, 7, 8, 9, "a", "b", "c", "d", "e", "f"), vol.Coerce(str)
-)
 
-CONFIG_SCHEMA = vol.Schema(
-    {
-        DOMAIN: vol.Schema(
-            {
-                cv.string: vol.Schema(
-                    {
-                        vol.Required(CONF_HOST): cv.string,
-                        vol.Optional(CONF_PORT, default=DEFAULT_PORT): cv.port,
-                        vol.Required(CONF_SWITCHES): vol.Schema(
-                            {RELAY_ID: SWITCH_SCHEMA}
-                        ),
-                    }
-                )
-            }
-        )
-    },
-    extra=vol.ALLOW_EXTRA,
-)
+async def async_setup(hass, config):
+    """Component setup, do nothing."""
+    hass.data.setdefault(DOMAIN, {})
+    return True
```

Comment:
If you leave `CONFIG_SCHEMA` in and add

```suggestion
    conf = config.get(DOMAIN)

    if not conf:
        return True

    hass.async_create_task(
        hass.config_entries.flow.async_init(
            DOMAIN,
            context={"source": SOURCE_IMPORT},
            data=conf
        )
    )

    return True
```

Import should be as simple as this in `config_flow.py`

```
    async def async_step_import(self, user_input):
        """Handle import."""
        return await self.async_step_user(user_input)
```

Reply:
How would this work when `CONFIG_SCHEMA` doesn't match the new `DATA_SCHEMA`?

Reply:
Ignore `CONF_SWITCHES`

Reply:
If you need a sample import test, you can borrow from `nut/test_config_flow.py`

Reply:
So I can't seem to get this working properly, there's something messing with the exception handling for `validate_input` but I can't figure out the cause. This does not work as is since each yaml config for this device can contain multiple devices, so I have it looping though and importing all entries.

Reply:
Here's more or less what I tried: https://github.com/jameshilliard/home-assistant/commit/3d5491f16bd51c8ce2c01860ee8d49f5da89715e

Reply:
You are pretty close on this, you shouldn't have to make any changes to the config flow beyond 

```
    async def async_step_import(self, user_input):
        """Handle import."""
        return await self.async_step_user(user_input)
```

Reply:
I tried that first, it didn't work.

Reply:
I updated the suggestion above since the schema is a bit different for this integration.

Reply:
Ok, I think that works, updated.